pages/cql/cms124_FHIR-7.2.000.cql

```diff
@@ -0,0 +1,127 @@
+valueset "Race": 'urn:oid:2.16.840.1.114222.4.11.836'
+valueset "Ethnicity": 'urn:oid:2.16.840.1.114222.4.11.837'
+valueset "Payer": 'urn:oid:2.16.840.1.114222.4.11.3591'
+valueset "Female": 'urn:oid:2.16.840.1.113883.3.560.100.2'
+valueset "Home Healthcare Services": 'urn:oid:2.16.840.1.113883.3.464.1003.101.12.1016'
+valueset "Hysterectomy with No Residual Cervix": 'urn:oid:2.16.840.1.113883.3.464.1003.198.12.1014'
+valueset "Office Visit": 'urn:oid:2.16.840.1.113883.3.464.1003.101.12.1001'
+valueset "Pap Test": 'urn:oid:2.16.840.1.113883.3.464.1003.108.12.1017'
+valueset "Preventive Care Services - Established Office Visit, 18 and Up": 'urn:oid:2.16.840.1.113883.3.464.1003.101.12.1025'
+valueset "Preventive Care Services-Initial Office Visit, 18 and Up": 'urn:oid:2.16.840.1.113883.3.464.1003.101.12.1023'
+valueset "HPV Test": 'urn:oid:2.16.840.1.113883.3.464.1003.110.12.1059'
+
+code "Congenital absence of cervix (disorder)": '37687000' from "SNOMEDCT:2017-09" display 'Congenital absence of cervix (disorder)'
+
+parameter "Measurement Period" Interval<DateTime>
+
+context Patient
+
+define "Ethnicity Categories":
+		flatten (
+        (
+            Patient.extension Extensions
+                where Extensions.url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity'
+        ) EthnicityExtensions
+            where EthnicityExtensions.extension.url = 'ombCategory'
+        return EthnicityExtensions.extension
+		)
+
+define "SDE Ethnicity":
+		"Ethnicity Categories" EthnicityExt
+				where FHIRHelpers.ToCode(EthnicityExt.value as FHIR.Coding) in "Ethnicity"
+		return FHIRHelpers.ToCode(EthnicityExt.value as FHIR.Coding).display
+
+		// DT: Not addressed in QI Core / FHIR
+// CRS: Use the "Payer" ValueSet to investigate what codes are expected and investigate whether an approach can be used with Coverage resource
+define "SDE Payer":
+	["Patient Characteristic Payer": "Payer"]
+
+define "Race Categories":
+		flatten (
+        (
+            Patient.extension Extensions
+                where Extensions.url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-race'
+        ) RaceExtensions
+            where RaceExtensions.extension.url = 'ombCategory'
+        return RaceExtensions.extension
+    )
+
+define "SDE Race":
+		"Race Categories" RaceExt
+				where FHIRHelpers.ToCode(RaceExt.value as FHIR.Coding) in "Race"
+		return FHIRHelpers.ToCode(RaceExt.value as FHIR.Coding).display
+		
+define "SDE Sex":
+		[Patient: gender in "ONC Administrative Sex"] P
+			return P.gender
+
+define "Numerator":
+	exists "Pap Test Within 3 Years"
+		or exists "Pap Test With HPV Within 5 Years"
+
+define "Denominator":
+	"Initial Population"
+
+define "Pap Test with Results":
+	[Observation: "Pap Test"] PapTest
+		where PapTest.value is not null
+
+// DT need to investigate more
+define "Pap Test Within 3 Years":
+	"Pap Test with Results" PapTest
+		where PapTest.relevantPeriod 3 years or less before end of "Measurement Period"

```

Comment:
relevantPeriod is a QDM element. Use FHIR Observation.effective element as discussed in mapping tables.